File: src/api/discussion-read.ts
import { Discussion, SessionHandle } from "~/models";
import { discussionMessages } from "./discussion-messages";

/**
 * Mark a discussion as read.
 * @remark Shortcut of `discussionMessages` but here we don't return anything.
 */
export const discussionRead = async (session: SessionHandle, discussion: Discussion) => {
  await discussionMessages(session, discussion, true);
};

File: library/src/commonMain/kotlin/ink/literate/pawnote/api/discussion-read.kt
package ink.literate.pawnote.api

import ink.literate.pawnote.models.Discussion
import ink.literate.pawnote.models.SessionHandle

/**
 * Mark a discussion as read.
 *
 * Shortcut of `discussionMessages` but here we don't return anything.
 */
suspend fun discussionRead(session: SessionHandle, discussion: Discussion) {
  discussionMessages(session, discussion, markAsRead = true)
}
